wandb/sdk/artifacts/artifacts_cache.py

```diff
@@ -181,6 +182,11 @@ def helper(mode: str = "w") -> Generator[IO, None, None]:
             try:
                 yield temp_file
                 temp_file.close()
+                # NamedTemporaryFile sets the file mode to 600 [1], we reset to the default.
+                # [1] https://stackoverflow.com/questions/10541760/can-i-set-the-umask-for-tempfile-namedtemporaryfile-in-python
+                umask_cmd = ("python", "-c", "import os; print(os.umask(22))")
```

Comment:
`python` might not be on `PATH`, use:
```suggestion
                umask_cmd = (sys.executable, "-c", "import os; print(os.umask(22))")
```